### PR TITLE
Adding shuffle array and select certain range of items

### PR DESCRIPTION
**Pull request by krehwell**
it would be cool if I can shuffle an array too since this is a library meant for having fun with array.

I also made a better options parameter for the `selectRandomArrayItem` following your issues on enhancement https://github.com/TechDevils/array-random-picker/issues/2 . 

now the function has a parameter that can be passed as an option either they want to show the key or they want to return some range number of random items selected

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,32 +3,41 @@ function switchDebug(){
     debug = !debug;
 }
 
-function randomNumberBetween(min, max) {
-    min = Math.ceil(min)
-    max = Math.floor(max)
-    var r = Math.random()
-    return Math.floor(r * (max - min + 1) + min)
-}
+function selectRandomArrayItem(input, options) {
```
Comment (Chadders13) [inline index.js]: could make options a global and then use the options input as a way to update the options.

then make the options an object 
```
{
    inculdeIndex: false,
    returnRange : 1
}

Comment (Chadders13) [reply index.js]: then that also would make it easier to extend later if needed

File: index.js [hunk 2]
```diff
@@ -3,32 +3,41 @@ function switchDebug(){
...
+    var includeIndex = false, range = 1;
 
-function selectRandomArrayItem(input, includeIndex) {
-    var inputLength = input.length - 1
-    var randomIndex = randomNumberBetween(0, inputLength)
+    if(arguments.length > 2)
+        return new Error("Error Too Many Params are Passed");
```
Comment (Chadders13) [inline index.js]: Nice one on the error handling

File: index.js [hunk 3]
```diff
@@ -3,32 +3,41 @@ function switchDebug(){
...
-        }
+
+    else if(typeof options === "boolean"){
+        range = 1;
+        includeIndex = options;
     }
 
-    return input[randomIndex];
+    else
+        return new Error("Unknown Error Has Occured During selectRandomArrayItem()");
```
Comment (Chadders13) [inline index.js]: Nice one on the error handling

**Review by Chadders13 — CHANGES_REQUESTED** [review]
Just got a slight change that should help to make it easier to add a repeating and none repeating random item later

File: index.js (hunk 2 above)
Comment (krehwell) [reply index.js]: glad if you like it, thanks

File: index.js (hunk 1 above)
Comment (krehwell) [reply index.js]: this new commit should fulfil your needs. refer https://github.com/TechDevils/array-random-picker/pull/3/commits/1d52e8ad0a831115a2774b39ac86e007c577b4a2

**Review by Chadders13 — APPROVED** [review]

**Pull request merged**